src/components/portfolio/DetailsMainBacktesting.tsx

```diff
@@ -0,0 +1,241 @@
+  const oneYearBack = new Date(today - 31556952000);
+  const [selectedFrom, setSelectedFrom] = React.useState<string>(
+    twoYearsBack.toISOString().split('T')[0]
+  );
+  const [selectedTo, setSelectedTo] = React.useState<string>(
+    oneYearBack.toISOString().split('T')[0]
+  );
+
+  const [helperText, setHelperText] = React.useState<string>('');
+  const [disabled, setDisabled] = React.useState<boolean>(false);
+  const [backtesting, setBacktesting] = React.useState<Backtesting>();
+  const [error, setError] = React.useState<Error | undefined>(undefined);
+  const isMounted = React.useRef(true);
+
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars
+  const fetch = async (from: number, to: number) => {
+    setError(undefined);
+    try {
+      // TODO get real token and id from useContext
+      const backTestingResponse = await API.backtesting(token, id, from, to);
+      if (isMounted.current) {
+        setBacktesting(backTestingResponse);
+      }
+    } catch (e) {
+      if (isMounted.current) {
+        setError(e);
+      }
+    }
+  };
+
+  useEffect(() => {
+    const from = Date.parse(selectedFrom);
+    const to = Date.parse(selectedTo);
+    if (to < from || Date.now() < to) {
+      setHelperText(t('portfolio.details.backtesting.dateHelperText'));
+    } else {
+      setHelperText('');
+      setDisabled(true);
+      // TODO do API call, call setBacktesting and then enable fields again
+      // TODO replace mock with actual api call
+      // fetch(from, to);
+      const backtestingInit: Backtesting = {
+        MDDMaxToMin: -65,
+        MDDInitialToMin: -65,
+        dateMax: '10.02.2019',
+        dateMin: '03.04.2020',
+        maxValue: 1250.55,
+        minValue: 512.67,
+        initialValue: 840.56,
+        bestYear: {
+          changeBest: 210.5,
+          yearBest: '2020',
+          growthRateBest: 10.5,
+        },
+        worstYear: {
+          changeWorst: -70.56,
+          yearWorst: '2019',
+          growthRateWorst: -5.6,
+        },
+        finalPortfolioBalance: 970.43,
+        CAGR: 5.42,
+        standardDeviation: 12.1,
+        sharpeRatio: 0.65,
+      };
+      setBacktesting(backtestingInit);
+
+      setTimeout(() => setDisabled(false), 3000);
+    }
+    return () => {
+      isMounted.current = false;
+    };
+  }, [selectedFrom, selectedTo, t]);
+
+  return (
+    <div className={classes.backtestingWrapper}>
+      <p className={classes.subtitle}>
+        {t('portfolio.details.backtesting.subheading')}
+      </p>
+      <div className={classes.datePicker}>
+        <form className={classes.form} noValidate>
+          <p className={classes.subtitle}>{t('portfolio.details.from')}:</p>
+          <TextField
+            id="dateFrom"
+            type="date"
+            disabled={disabled}
+            value={selectedFrom}
+            variant="outlined"
+            onChange={(e) => setSelectedFrom(e.target.value)}
```

Comment:
How would this differ? I want the value from the text field not from the form.

Reply:
`e.target` is the element that triggers the event. Usually it's the input element, but it could also be a child element of the input which might not contain the value that you want. Because we don't know the internals of the input element (there may be inconsistencies across browsers and Material-UI could change it in a new version), using `e.target` is unsafe.

`e.currentTarget` is the element that the event listener is attached to. It's always the input element in this case. Accessing its value is safe.